File: android_samples_java/19.2 SukritiTest/app/src/main/java/com/sukriti/sukrititest/ThreeFragment.java
package com.sukriti.sukrititest;

import android.app.Activity;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class ThreeFragment extends Fragment {

    AppCompatImageView decrementImgbtn;
    CallbackInterface callbackInterface;

    @Override
    public void onAttach(@NonNull Activity activity) {
        super.onAttach(activity);

        callbackInterface = (SecondActivity) activity;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {
        View view = inflater.inflate(R.layout.fragment_three, container, false);

        decrementImgbtn = view.findViewById(R.id.decrementImgbtn);
        decrementImgbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                callbackInterface.decrement();
            }
        });

        return view;
    }
}

File: android_samples_kotlin/08 SukritiTest/app/src/main/java/com/sukriti/sukrititest/ThreeFragment.kt
package com.sukriti.sukrititest

import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatImageView
import androidx.fragment.app.Fragment
import com.sukriti.sukrititest.databinding.FragmentThreeBinding

class ThreeFragment : Fragment() {

    private var _binding: FragmentThreeBinding? = null
    private val binding get() = _binding!!
    var callbackInterface: CallbackInterface? = null

    override fun onAttach(activity: Activity) {
        super.onAttach(activity)

        callbackInterface = activity as SecondActivity
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {

        _binding = FragmentThreeBinding.inflate(inflater, container, false)

        binding.decrementImgbtn!!.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                callbackInterface!!.decrement()
            }
        })

        return binding.root
    }
}

